docx/oxml/numbering.py

```diff
@@ -139,11 +139,21 @@ def get_abstractNum(self, numId):
             if el.abstractNumId == abstractNum_id:
                 return el
 
-    def get_lvl_for_p(self, p, styles_cache):
+    def get_lvl_from_props(self, p, styles_cache=None):
         """
-        Gets the formatting based on current paragraph indentation level.
+        Gets the formatting based on current paragraph indentation level defined in paragraph styles.
+        If ``styles_cache`` is not None then level from style formating is fetched otherwise
+        level from direct paragraph formating is used.
         """
-        numPr = p.pPr.get_numPr(styles_cache)
+        try:
+            if styles_cache:
+                numPr = p.pPr.get_style_numPr(styles_cache)
+            else:
+                numPr = p.pPr.numPr
+            if numPr is None:
+                raise AttributeError
```

Comment:
would it make sense to replace this with the rest of the code in this function, so if at some point `AttributeError` is raised `None` shall be returned. Note that you can just type `pass` instead of `return None` because `None` shall be returned by default if a function doesn't return anything.